lib/std/fs/Dir.zig

```diff
@@ -118,6 +137,13 @@ pub const Iterator = switch (builtin.os.tag) {
                     continue :start_over;
 
                 // Solaris dirent doesn't expose d_type, so we have to call stat to get it.
```

Comment:
I think this comment belongs with the `posix.fstat()` call on L147, so it needs to be moved after your new code?

Reply:
For me this was a comment about why we need additional code/syscall. Given that my addition causes the stat call to be skipped, I've added it below the comment.

But I can move the comment, if you/other think it fits better. Or maybe add a comment inside my `if` like `// Skip the additional stat call`

Reply:
Oh, I see how you're reading it now.  That makes sense.  This is fine as-is.